### PR TITLE
Add binary generation workflow

### PR DESCRIPTION
**Pull request by ErebusZ**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,7 +31,6 @@ MANIFEST
 #  Usually these files are written by a python script from a template
 #  before PyInstaller builds the exe, so as to inject date/other infos into it.
 *.manifest
-*.spec
```
Comment (mohsinenar) [inline .gitignore:34]: Why?

**Review by mohsinenar — CHANGES_REQUESTED** [review]

File: main.spec [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# -*- mode: python ; coding: utf-8 -*-
+
+
+a = Analysis(
+    ['/home/azouiten/ostorlab/src/ostorlab/main.py'],
```
[the remark this answers is not in the document]
Comment (ErebusZ) [reply main.spec:5]: This currently draft Pr is a testing ground for the workflow.

File: .gitignore (hunk 1 above)
Comment (ErebusZ) [reply .gitignore:34]: .spec file would give the ability to further configure PyInstaller.

Comment (3asm) [reply .gitignore:34]: You don't have a spec file in your PR. Why remove it then?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      run: |
+        git submodule init
+        git submodule update
+
+    - name: Install dependencies.
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[scanner]
+        pip install .[agent]
+        python -m pip install -e .
```
Comment (3asm) [inline .github/workflows/release.yaml:30]:
```suggestion
        pip install .
        pip install .[scanner]
        pip install .[agent]
```

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        pip install pyinstaller
+
+    - name: Debug dependencies.
+      run: pip freeze
```
Comment (3asm) [inline .github/workflows/release.yaml:34]: Why?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 4 above)
Comment (ErebusZ) [reply .github/workflows/release.yaml:34]: just for debugging purposes

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+on:
+  push:
```
Comment (mohsinenar) [inline .github/workflows/release.yaml:2]: You need to change this to release! not push.

**Review by mohsinenar — DISMISSED** [review]

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
```
Comment (benyissa) [inline .github/workflows/release.yaml:21]:
```suggestion
          cache: 'pip'
```

File: .github/workflows/release.yaml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install .
+        pip install .[scanner]
+        pip install .[agent]
+        pip install pyinstaller
+
+    - run: |
+        pyinstaller --name oxo --onefile --add-data="./*:." --hidden-import alembic --hidden-import docker --hidden-import click src/ostorlab/main.py
+
```
Comment (3asm) [inline .github/workflows/release.yaml:32]: Why do you need to set this manually. How do I know this list is complete?

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**